src/Trials.jl

```diff
@@ -20,7 +20,7 @@ $(TYPEDFIELDS)
 - `loss` : the float returned by evaluating the user's function
 """
 mutable struct Trial
-    hyperparams::Dict{Symbol, T} where T
+    hyperparams::Any
```

Comment:
Once the calls were free'd up this way, expecting a dict wasn't possible any more. In fact inferring anything about the generated parameters is impossible. As tests show, you could have directly a float, an array of floats, an array of dicts, an array of anything, a dict, and so on. However, I believe this is the same level of flexibility as afforded by hyperopt. This was also necessary for being able to define a space which is composed of a dict and is flat once generated

Reply:
Makes sense to me, but maybe add a comment in the code? i.e. what the code expects the content of hyperparams to be.

Reply:
Can do but the problem is that it can literally genuinely be (almost) anything. Int? sure. String? sure. Dict? sure. array of dicts/ints/strings/mixed? sure

Reply:
yea true